### PR TITLE
feat: Add api key authentication through attributes

### PR DESCRIPTION
**Pull request by janlesage**

### CODE REVIEW

**Review by rikdepeuter — APPROVED** [review]

File: src/RoadRegistry.BackOffice.Api/Changes/ChangeFeedController.cs [hunk 1]
```diff
@@ -17,10 +19,11 @@ namespace RoadRegistry.BackOffice.Api.Changes;
 using NodaTime;
 using NodaTime.Text;
 
-[ApiVersion("1.0")]
-[AdvertiseApiVersions("1.0")]
+[ApiVersion(Version.Current)]
```
Comment (lvermeulen) [inline src/RoadRegistry.BackOffice.Api/Changes/ChangeFeedController.cs:22]: why change from 1.0 to current?

File: src/RoadRegistry.BackOffice.Api/Infrastructure/Configuration/FeatureToggleOptions.cs [hunk 2]
```diff
@@ -4,5 +4,6 @@ public sealed class FeatureToggleOptions
     {
         public const string ConfigurationKey = "FeatureToggles";
         public bool UseFeatureCompare { get; set; }
+        public bool UseApiKeyAuthentication { get; set; }
```
Comment (lvermeulen) [inline src/RoadRegistry.BackOffice.Api/Infrastructure/Configuration/FeatureToggleOptions.cs:7]: this is the one feature that doesn't need a feature flag :-)

File: src/RoadRegistry.BackOffice.Api/RoadRegistry.BackOffice.Api.csproj [hunk 3]
```diff
@@ -9,9 +9,10 @@
     <EnableDefaultContentItems>false</EnableDefaultContentItems>
     <Nullable>disable</Nullable>
     <DocumentationFile>RoadRegistry.BackOffice.Api.xml</DocumentationFile>
-    <NoWarn>1701;1702;1705;1591</NoWarn>
+    <NoWarn>1701;1702;1705;1591;8632</NoWarn>
     <Platforms>AnyCPU;x64;x86</Platforms>
     <ErrorOnDuplicatePublishOutputFiles>false</ErrorOnDuplicatePublishOutputFiles>
+    <UserSecretsId>245c5c1d-6913-4f11-80d6-bc3cb6bae995</UserSecretsId>
```
Comment (lvermeulen) [inline src/RoadRegistry.BackOffice.Api/RoadRegistry.BackOffice.Api.csproj:15]: remove this line

File: src/RoadRegistry.BackOffice.Api/Infrastructure/Configuration/FeatureToggleOptions.cs (hunk 2 above)
Comment (janlesage) [reply src/RoadRegistry.BackOffice.Api/Infrastructure/Configuration/FeatureToggleOptions.cs:7]: Indeed, however for local development, we can shut it down in appsettings.development.json. We can remove if need be?

**Pull request merged**